magma/lib/magma/attributes/file.rb

```diff
@@ -144,8 +144,10 @@ def self.type_bulk_load_hook(loader, project_name, attribute_copy_revisions)
 
     private
 
-    def filename(record_name, path)
+    def filename(record_name, path, original_filename=nil)
       ext = path ? ::File.extname(path) : ''
+      original_ext = original_filename ? ::File.extname(original_filename) : ''
+      ext = original_ext if ext.empty?
```

Comment:
Uses the original file name's extension if available. Otherwise temporary upload URLs have no knowledge of the right extensions and would just appear as `.dat`, which is confusing for users.